### PR TITLE
Extended to set custom Server ID

### PR DESCRIPTION
**Pull request by madnuttah**
Script and yaml files edited, I hope it's fine.

### CODE REVIEW

**Comment by madnuttah** [comment]
The concerns regarding the `wait` command could be circumvented by (mis)using a healthcheck-script like I did [here](https://github.com/madnuttah/unbound-docker-stats/blob/main/README.md#instructions)

**Comment by madnuttah** [comment]
My apologies, apparently this is new to me.

**Comment by bazzani** [comment]
Hi @robinmanuelthiel and @madnuttah 

I have been looking at this repo too and I want to introduce a similar change to supply a hostname (with `-o` argument) to the `speedtest` command optionally, and I plan to open a PR soon from my fork.

---

Concerning this PR, I tested your changes locally @madnuttah and I observed a problem where the docker container fails if I do not specify the server ID.  I think it would be good to make this ability to add a server ID optional, and if it is missing the the regular `speedtest` CLI command should be invoked without the `-s` argument.

Container logs:

```
Running a Speed Test...
2024-05-30T07:41:46.102927881Z Thu May 30 07:41:46 2024
2024-05-30T07:41:46.107768923Z speedtest: Missing argument for option: -s
2024-05-30T07:41:46.149092423Z Unable to retrieve JSON results from speedtest
2024-05-30T07:41:46.149114881Z Running next test in 1800s...
2024-05-30T07:41:46.149116256Z 
```

---

p.s. the similar change I made can be found at
- https://github.com/bazzani/speedtest/commit/2033353a1b671a3be0b7de876376f49c2cbe9568

**Comment by robinmanuelthiel** [comment]
Yeah, if we introduce Server ID and URL, I would like to see them as _optional_ parameters as well. This implementation here requires them.

File: speedtest.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (robinmanuelthiel) [inline speedtest.sh:1]: Any reason why you changed from shell to bash?

File: speedtest.sh [hunk 2]
```diff
@@ -15,24 +16,31 @@ run_speedtest()
...
+    echo "Your test server is $SERVER (ID $SERVERID) in $SERVERLOCATION."
     echo "Your download speed is $(($DOWNLOAD  / 125000 )) Mbps ($DOWNLOAD Bytes/s)."
     echo "Your upload speed is $(($UPLOAD  / 125000 )) Mbps ($UPLOAD Bytes/s)."
     echo "Your ping is $PING ms."
 
     # Save results in the database
     if $DB_SAVE; 
     then
-        echo "Saving values to database..."
+        echo "Saving values to database..."
```
Comment (robinmanuelthiel) [inline speedtest.sh:37]: Pls remove added sapces here

File: speedtest.sh [hunk 3]
```diff
@@ -15,24 +16,31 @@ run_speedtest()
...
     then
-        echo "Saving values to database..."
+        echo "Saving values to database..."      
         curl -s -S -XPOST "$DB_HOST/write?db=$DB_NAME&precision=s&u=$DB_USERNAME&p=$DB_PASSWORD" \
             --data-binary "download,host=$HOSTNAME value=$DOWNLOAD $DATE"
         curl -s -S -XPOST "$DB_HOST/write?db=$DB_NAME&precision=s&u=$DB_USERNAME&p=$DB_PASSWORD" \
             --data-binary "upload,host=$HOSTNAME value=$UPLOAD $DATE"
         curl -s -S -XPOST "$DB_HOST/write?db=$DB_NAME&precision=s&u=$DB_USERNAME&p=$DB_PASSWORD" \
-            --data-binary "ping,host=$HOSTNAME value=$PING $DATE"
+            --data-binary "ping,host=$HOSTNAME value=$PING $DATE"
```
Comment (robinmanuelthiel) [inline speedtest.sh:43]: Pls remove added sapces here

**Review by robinmanuelthiel — CHANGES_REQUESTED** [review]
Thank you very much for this change! I would like to see the server ID as an _optional_ parameter, though. Can you make it optional yourself, or do you want help with that?

**Comment by robinmanuelthiel** [comment]
> Hi @robinmanuelthiel and @madnuttah
> 
> I have been looking at this repo too and I want to introduce a similar change to supply a hostname (with `-o` argument) to the `speedtest` command optionally, and I plan to open a PR soon from my fork.
> 
> Concerning this PR, I tested your changes locally @madnuttah and I observed a problem where the docker container fails if I do not specify the server ID. I think it would be good to make this ability to add a server ID optional, and if it is missing the the regular `speedtest` CLI command should be invoked without the `-s` argument.
> 
> Container logs:
> 
> ```
> Running a Speed Test...
> 2024-05-30T07:41:46.102927881Z Thu May 30 07:41:46 2024
> 2024-05-30T07:41:46.107768923Z speedtest: Missing argument for option: -s
> 2024-05-30T07:41:46.149114881Z Running next test in 1800s...
> 2024-05-30T07:41:46.149116256Z 
> ```
> 
> p.s. the similar change I made can be found at
> 
> * [bazzani@2033353](https://github.com/bazzani/speedtest/commit/2033353a1b671a3be0b7de876376f49c2cbe9568)

This looks promising. Can you create a PR from your repo back into this one?

**Comment by bazzani** [comment]
> > p.s. the similar change I made can be found at
> > 
> > * [bazzani@2033353](https://github.com/bazzani/speedtest/commit/2033353a1b671a3be0b7de876376f49c2cbe9568)
> 
> This looks promising. Can you create a PR from your repo back into this one?

Hi @robinmanuelthiel, may I ask you to please clarify that you are referring to my suggested change?
- https://github.com/bazzani/speedtest/commit/2033353a1b671a3be0b7de876376f49c2cbe9568

If so, I can work on this, and also add support for the `-s` flag at the same time
**although providing them must be mutually exclusive or _this causes an error_ when running the speedtest cli with both `-o` and `-s` provided** 👇 

- ```sh
  speedtest --accept-license --accept-gdpr -f json -s 28910 -o lg-tok.fdcservers.net
  [2024-06-09 21:50:15.193] [error] Only one server selection can be specified
  ```

**Comment by robinmanuelthiel** [comment]
Yes, exactly. To me it's important, that I also works without specifying any server name or IP.

**Comment by bazzani** [comment]
> Yes, exactly. To me it's important, that I also works without specifying any server name or IP.

100%, I will open the PR soon enough, after giving it a thorough test locally.

**Comment by madnuttah** [comment]
Hey there and sorry for my late reply. I must confess this was quite rushed. It was working so I submitted the PR without any further tests, as I wrote it did what I wanted. I'm sorry if it didn't meet the quality you've expected. As I'm doing much better again, I'm getting a take on this again.

**Comment by robinmanuelthiel** [comment]
@bazzani Not sure, if I like the build part in the Docker Compose file, as I wanted to give an example, how to use the container easily in Docker Compose file. Building it yourself, would make it harder. But we could maybe add another examle file for self-building or just use both, `image` and `build` https://docs.docker.com/compose/compose-file/build/#using-build-and-image

**Comment by robinmanuelthiel** [comment]
> Hey there and sorry for my late reply. I must confess this was quite rushed. It was working so I submitted the PR without any further tests, as I wrote it did what I wanted. I'm sorry if it didn't meet the quality you've expected. As I'm doing much better again, I'm getting a take on this again.

Hey no worries - thanks for the contribution. We are working on this together! 💪

**Comment by bazzani** [comment]
> @bazzani Not sure, if I like the build part in the Docker Compose file, as I wanted to give an example, how to use the container easily in Docker Compose file. Building it yourself, would make it harder. But we could maybe add another examle file for self-building or just use both, `image` and `build` https://docs.docker.com/compose/compose-file/build/#using-build-and-image

I see what you mean @robinmanuelthiel, thanks for the input and helping me consider the other optionsyou proposed.
I think I was perhaps a tad too deep in the developer mindset, and not enough in a consumer (_of the service_) mindset.

I will review that link you sent and play about a bit with the image and build approach.

**Comment by bazzani** [comment]
> Hey there and sorry for my late reply. I must confess this was quite rushed. It was working so I submitted the PR without any further tests, as I wrote it did what I wanted. I'm sorry if it didn't meet the quality you've expected. As I'm doing much better again, I'm getting a take on this again.

Thank you for the PR @madnuttah, I do appreciate your changes as it helped me to understand the project more, and made me more curious as to how we can extend it by working together. **OSS is the best**!

File: README.md [hunk 4]
```diff
@@ -28,6 +28,7 @@ Your ping is 6.223 ms.
 | `DB_NAME`            | `speedtest`             | InfluxDB Database name            |
 | `DB_USERNAME`        | `admin`                 | InfluxDB Username                 |
 | `DB_PASSWORD`        | `password`              | InfluxDB Password                 |
+| `TESTSERVER_ID`      | `61186`              | ServerID, you get this by inspecting your ISP entry [Ookla Speedtest Website](speedtest.net)                  |
```
[the remark this answers is not in the document]
Comment (madnuttah) [reply README.md:31]: Before I start, some questions. Do you prefer having a generic string here or do you have a completely different approach in mind? I just picked one. I guess describing better how to get a ServerID is also necessary here.

File: speedtest.sh (hunk 2 above)
Comment (madnuttah) [reply speedtest.sh:37]: Will do

File: speedtest.sh (hunk 3 above)
Comment (madnuttah) [reply speedtest.sh:43]: Also here

**Comment by madnuttah** [comment]
@bazzani it seems like you invested way more time into this and the script looks better already than what I have here right now. I'll close this, also because of a lack of time. Sorry, I shouldn't have done this that time.

**Comment by robinmanuelthiel** [comment]
> Sorry, I shouldn't have done this that time.

I disagree! Your effort kicked it all off. Can't wait to see this feature here soon. Thanks for your work @madnuttah! Keep it coming. (both feedback, issues and code contributions)

**Comment by robinmanuelthiel** [comment]
@bazzani, are you still planning to create a PR for the Server ID?

**Comment by bazzani** [comment]
> @bazzani, are you still planning to create a PR for the Server ID?

Hi @robinmanuelthiel i am indeed. I started working on it last week and got stuck on a few shell scripting logic challenges (due to the two mutually exclusive optional flags). I then got sick on Friday so I have not had a chance to complete it yet.

I will look at it tomorrow and get something in a PR for your review.

**Comment by bazzani** [comment]
FYI @robinmanuelthiel @madnuttah #19

**Pull request closed without merging**